soularr.py

```diff
@@ -240,99 +243,236 @@ def verify_filetype(file,allowed_filetype):
+    if is_blacklisted(album_title):
+       return False, artist_name 
+
+    if len(album_title) == 1:
+        query = '"' + artist_name + '"' + " " + album_title
+    else:
+        query = '"' + artist_name + '"' + " " + album_title if config.getboolean('Search Settings', 'album_prepend_artist', fallback=False) else album_title
+
+    logger.info(f"Searching album: {query}")
     search = slskd.searches.search_text(searchText = query,
                                         searchTimeout = config.getint('Search Settings', 'search_timeout', fallback=5000),
                                         filterResponses = True,
                                         maximumPeerQueueLength = config.getint('Search Settings', 'maximum_peer_queue', fallback=50),
                                         minimumPeerUploadSpeed = config.getint('Search Settings', 'minimum_peer_upload_speed', fallback=0))
 
-    track_num = len(tracks)
-
-    time.sleep(5)
+    time.sleep(10)
     while True:
-        if slskd.searches.state(search['id'])['state'] != 'InProgress':
+        state = slskd.searches.state(search['id'],False)['state']
+        pprint.pprint(state)
+        if state != 'InProgress':
             break
         time.sleep(1)
 
     logger.info(f"Search returned {len(slskd.searches.search_responses(search['id']))} results")
 
     dir_cache = {}
+    search_cache = {}
 
     for result in slskd.searches.search_responses(search['id']):
         username = result['username']
-        if not username in dir_cache:
+        if username not in dir_cache:
             dir_cache[username] = {}
         logger.info(f"Truncating directory count of user: {username}")
         init_files = result['files']
-        for file in init_files:                   
+        for file in init_files:
             file_dir = file['filename'].rsplit('\\',1)[0]
             for allowed_filetype in allowed_filetypes:
                 if verify_filetype(file, allowed_filetype):
                     if allowed_filetype not in dir_cache[username]:
                         dir_cache[username][allowed_filetype] = []
                     if file_dir not in dir_cache[username][allowed_filetype]:
                         dir_cache[username][allowed_filetype].append(file_dir)
-
+    
     for allowed_filetype in allowed_filetypes:
         logger.info(f"Serching for matches with selected attributes: {allowed_filetype}")
-
-        for username in dir_cache:
-
-            if not allowed_filetype in dir_cache[username]:
-                continue
-            logger.info(f"Parsing result from user: {username}")
-            for file_dir in dir_cache[username][allowed_filetype]:
-
-                try:
-                    directory = slskd.users.directory(username = username, directory = file_dir)
-                except: 
+        releases = lidarr.get_album(album_id)['releases']
+        num_releases = len(releases)
+        for i in range(0,num_releases):
```

Comment:
```suggestion
        for _ in range(0,num_releases):
```
I don't believe `i` is ever used here.